cmd/krew/cmd/list.go

```diff
@@ -56,20 +134,55 @@ Remarks:
 				return nil
 			}
 
-			// print table
-			var rows [][]string
-			for p, version := range plugins {
-				rows = append(rows, []string{p, version})
+			if *output.OutputFormat == "wide" {
+				// print table
+				var rows [][]string
+				for p, version := range plugins {
+					rows = append(rows, []string{p, version})
+				}
+				rows = sortByFirstColumn(rows)
+				return printTable(os.Stdout, []string{"PLUGIN", "VERSION"}, rows)
 			}
-			rows = sortByFirstColumn(rows)
-			return printTable(os.Stdout, []string{"PLUGIN", "VERSION"}, rows)
+
+			if inArray(*output.OutputFormat, output.AllowedFormats()) {
+				objs := []Entry{}
+				for plugin, version := range plugins {
+					obj := Entry{plugin, version}
+					objs = append(objs, obj)
+				}
+				objs = sortByName(objs)
```

Comment:
plugins is a map, as far as I know a map can't be sorted?

Reply:
yeah, but we can do a sortedPluginNames and use it in nearly all these printers, instead of sorting in multiple places etc?

Reply:
I'll try and figure something out. Part of the problem here is that `names` and `wide` break out of the printer implementations. The other part is that they use `[][]string` instead of `Entry` as the data structure.

Reply:
I can experiment with [NamesFlag](https://github.com/kubernetes/cli-runtime/blob/master/pkg/genericclioptions/name_flags.go) but it won't behave identically to `names` / redirect output without a flag.

The code for [PrintObj](https://github.com/kubernetes/cli-runtime/blob/master/pkg/printers/name.go#L110) requires there to be a non-empty kind and prints it as the prefix to all the plugin names. Output would look like this.

```
plugin/cssh
plugin/krew
plugin/tail
```

Reply:
> I can experiment with [NamesFlag](https://github.com/kubernetes/cli-runtime/blob/master/pkg/genericclioptions/name_flags.go) [](/kubernetes/cli-runtime/blob/HEAD@{2019-11-05T20:34:05Z}/pkg/genericclioptions/name_flags.go) but it won't behave identically to `names` / redirect output without a flag.

makes sense, I just realized. let's leave -o-=name out for now then.

Reply:
FWIW for sorting we can do this:

- make `installation.ListInstalledPlugins` return a `PluginList` type which is an alias to map[string]string

- implement a `SortedNames() []string` method on `PluginList` type –and use it here.

Reply:
I've left `-o=name` in just to leave an option of mimicking redirect output for terminal users.

Want me to remove it?

Reply:
We can keep -o=name it’s cool with me. 
Just need to write tests for it, too.